src/cli/cli_srp_server.cpp

```diff
@@ -0,0 +1,246 @@
+    VerifyOrExit(command != nullptr);
+
+    error = (this->*command->mHandler)(aArgsLength, aArgs);
+
+exit:
+    return error;
+}
+
+otError SrpServer::ProcessDomain(uint8_t aArgsLength, char *aArgs[])
+{
+    otError error = OT_ERROR_NONE;
+
+    if (aArgsLength > 1)
+    {
+        SuccessOrExit(error = otSrpServerSetDomain(mInterpreter.mInstance, aArgs[1]));
+    }
+    else
+    {
+        mInterpreter.OutputLine(otSrpServerGetDomain(mInterpreter.mInstance));
+    }
+
+exit:
+    return error;
+}
+
+otError SrpServer::ProcessEnable(uint8_t aArgsLength, char *aArgs[])
+{
+    OT_UNUSED_VARIABLE(aArgsLength);
+    OT_UNUSED_VARIABLE(aArgs);
+
+    otSrpServerSetEnabled(mInterpreter.mInstance, /* aEnabled */ true);
+
+    return OT_ERROR_NONE;
+}
+
+otError SrpServer::ProcessDisable(uint8_t aArgsLength, char *aArgs[])
+{
+    OT_UNUSED_VARIABLE(aArgsLength);
+    OT_UNUSED_VARIABLE(aArgs);
+
+    otSrpServerSetEnabled(mInterpreter.mInstance, /* aEnabled */ false);
+
+    return OT_ERROR_NONE;
+}
+
+otError SrpServer::ProcessLease(uint8_t aArgsLength, char *aArgs[])
+{
+    otError  error = OT_ERROR_NONE;
+    uint32_t minLease;
+    uint32_t maxLease;
+    uint32_t minKeyLease;
+    uint32_t maxKeyLease;
+
+    VerifyOrExit(aArgsLength == 5, error = OT_ERROR_INVALID_ARGS);
+    SuccessOrExit(error = Utils::CmdLineParser::ParseAsUint32(aArgs[1], minLease));
+    SuccessOrExit(error = Utils::CmdLineParser::ParseAsUint32(aArgs[2], maxLease));
+    SuccessOrExit(error = Utils::CmdLineParser::ParseAsUint32(aArgs[3], minKeyLease));
+    SuccessOrExit(error = Utils::CmdLineParser::ParseAsUint32(aArgs[4], maxKeyLease));
+
+    error = otSrpServerSetLeaseRange(mInterpreter.mInstance, minLease, maxLease, minKeyLease, maxKeyLease);
+
+exit:
+    return error;
+}
+
+otError SrpServer::ProcessHost(uint8_t aArgsLength, char *aArgs[])
+{
+    OT_UNUSED_VARIABLE(aArgs);
+
+    otError                error = OT_ERROR_NONE;
+    const otSrpServerHost *host;
+
+    VerifyOrExit(aArgsLength <= 1, error = OT_ERROR_INVALID_ARGS);
+
+    host = nullptr;
+    while ((host = otSrpServerGetNextHost(mInterpreter.mInstance, host)) != nullptr)
+    {
+        const otIp6Address *addresses;
+        uint8_t             addressesNum;
+        bool                isDeleted = otSrpServerHostIsDeleted(host);
+
+        mInterpreter.OutputLine(otSrpServerHostGetFullName(host));
+        mInterpreter.OutputLine(Interpreter::kIndentSize, "deleted: %s", isDeleted ? "true" : "false");
+        if (isDeleted)
+        {
+            continue;
+        }
+
+        mInterpreter.OutputSpaces(Interpreter::kIndentSize);
+        mInterpreter.OutputFormat("addresses: [");
+
+        addresses = otSrpServerHostGetAddresses(host, &addressesNum);
+
+        for (uint8_t i = 0; i < addressesNum; ++i)
+        {
+            mInterpreter.OutputIp6Address(addresses[i]);
+            if (i < addressesNum - 1)
+            {
+                mInterpreter.OutputSpaces(2);
```

Comment:
nit: be consistent with below Line#219
```suggestion
                mInterpreter.OutputFormat(", ");
```